packages/grafana-ui/src/components/uPlot/geometries/EventsCanvas.tsx

```diff
@@ -1,18 +1,18 @@
+import { DataFrame } from '@grafana/data';
 import React, { useMemo } from 'react';
-import { DataFrame, DataFrameView } from '@grafana/data';
 import { usePlotContext } from '../context';
+import { useRefreshAfterGraphRendered } from '../hooks';
 import { Marker } from './Marker';
 import { XYCanvas } from './XYCanvas';
-import { useRefreshAfterGraphRendered } from '../hooks';
 
-interface EventsCanvasProps<T> {
+interface EventsCanvasProps {
   id: string;
   events: DataFrame[];
-  renderEventMarker: (event: T) => React.ReactNode;
-  mapEventToXYCoords: (event: T) => { x: number; y: number } | undefined;
+  renderEventMarker: (event: DataFrame, index: number) => React.ReactNode;
+  mapEventToXYCoords: (event: DataFrame, index: number) => { x: number; y: number } | undefined;
```

Comment:
The `event` name for dataFrame is misleading should probably be `dataFrame`. Before it meant single row in a dataFrame of events, but then I would confuse it with DOM click event or something like that.

Reply:
makes sense.